devnet-sdk/system2/faucet.go

```diff
@@ -0,0 +1,22 @@
+package system2
+
+type FaucetID idWithChain[string]
+
+func (id FaucetID) String() string {
+	return idWithChain[string](id).string("Faucet")
+}
+
+func (id FaucetID) MarshalText() ([]byte, error) {
+	return idWithChain[string](id).marshalText("Faucet")
+}
+
+func (id *FaucetID) UnmarshalText(data []byte) error {
+	return (*idWithChain[string])(id).unmarshalText("Faucet", data)
+}
+
+type Faucet interface {
+	Common
+	ID() FaucetID
+	// NewUser creates a new pre-funded user account
+	NewUser() Wallet
```

Comment:
At the moment, the `devnet-sdk` receives a system/set of systems, rather than creating them. It can only return wallets it got from the system (e.g. a list of refunded accounts created by `anvil`), but it cannot "create" new funded accounts.

The general idea is that the systems are defined before devnet-sdk takes over which gives much more flexibility as to how and where the systems are defined (could be a local network, could be `op-mainnet`).

So the `Faucet` could work if there was a Faucet system component that devnet-sdk could talk to, otherwise it's up to the creator of the system to list a bunch of accounts with funds